src/fetch_family_allowance.py

```diff
@@ -0,0 +1,91 @@
+import datetime
+import warnings
+import requests
+import zipfile
+from urllib.request import urlretrieve
+import os
+import pandas
+import fileinput
+from sqlalchemy import create_engine
+import sys
+import shutil
+
+STARTING_YEAR = 2011
+
+NOW = datetime.datetime.now()
+
+SERVER = 'http://arquivos.portaldatransparencia.gov.br/'
+PATH = 'downloads.asp?a={}&m={:02d}&consulta=BolsaFamiliaFolhaPagamento'
+outputfile = 'data_family_allowance.csv'
+BASE_DATA_DIR = '../data/'
+TEMP_DATA = "../data/temp_data/"
+
+columns = {   'UF': 'FEDERAL_UNIT',
+              'CódigoSIAFIMunicípio': 'CODE_SIAFI_TONW',
+              'NomeMunicípio': 'TOWN_NAME',
+              'CódigoFunção': 'FUNCTION_CODE',
+              'CódigoSubfunção': 'SUBFUNCTION_CODE',
+              'CódigoPrograma': 'PROGRAM_CODE',
+              'CódigoAção': 'ACTION_CODE',
+              'NISFavorecido': 'FAVORED_NIS',
+              'NomeFavorecido': 'FAVORED_NAME',
+              'Fonte-Finalidade': 'SOURCE_FINALITY',
+              'ValorParcela': 'PARCEL_VALUE',
+              'MêsCompetência': 'MONTH_COMPETENCE',
+          }
+
+def dlProgress(count, blockSize, totalSize):
+  percent = int(count*blockSize*100/totalSize)
+  sys.stdout.write("\r" + "progress" + "...%d%%" % percent)
+  sys.stdout.flush()
+
+def urls():
+    for year in range(STARTING_YEAR, NOW.year + 1):
+        for month in range(1, 13):
+            if datetime.datetime(year, month, NOW.day) <= NOW:
+                url = SERVER + PATH.format(year, month)
+                req = requests.head(url)
+
+                # looks like some devs are not familiar with 404 status
+                if req.headers.get('Content-Type') == 'application/x-download':
+                    yield (url,year,month)
+
+                else:
+                    msg = 'Data from {:02d}/{} could not be located at: {}'
+                    warnings.warn(msg.format(month, year, url))
+
+def concat_csv_file(path_file):
```

Comment:
Why do we need this function? `urlretrive` already saves the file into the disk, right? The code is not clear, the lack of comments/docstrings and generci variables names (`i` and `j`) makes it really difficult to grasp it's idea.

Maybe a better strategy is to write three function and keep they simple:
1. `save_file` just wrapping `urlretrieve`
2. `unzip_file` (maybe merge it with the previous function, but anyway…) to uncompress the download file
3. `translate_headers` that would just read the first line and rewrite it